components/net/fetch/methods.rs

```diff
@@ -357,29 +357,29 @@ pub async fn main_fetch(
             // request's current URL's origin is same origin with request's origin, and request's
             // response tainting is "basic"
             if (same_origin && request.response_tainting == ResponseTainting::Basic) ||
-                // request's current URL's scheme is "data"
-                current_scheme == "data" ||
+                // request's current URL's scheme is "data" or is a fetchable custom protocol
+                context.protocols.is_fetchable(current_scheme) ||
```

Comment:
Hrm. So "request's current URL's scheme is "data"" here is a quote from is a quote from the specification so I don't think it's correct to just modify the text here. We need to explain why we deviate from the specification in this case with a comment that's clearly not spec text.

Reply:
I could also bring back that line if the performance isn't a problem, it's a bit more clear to me as well